src/Sylius/Component/Core/spec/Promotion/Action/UnitFixedDiscountPromotionActionCommandSpec.php

```diff
@@ -98,7 +109,63 @@ function it_applies_a_fixed_discount_on_every_unit_in_order(
         $unit1->addAdjustment($promotionAdjustment1)->shouldBeCalled();
         $unit2->addAdjustment($promotionAdjustment2)->shouldBeCalled();
 
-        $this->execute($order, ['amount' => 500], $promotion);
+        $this->execute($order, ['base_amount' => 500], $promotion);
+    }
+
+    function it_applies_a_fixed_discount_in_defined_currency_on_every_unit_in_order(
+        FactoryInterface $adjustmentFactory,
+        FilterInterface $priceRangeFilter,
+        FilterInterface $taxonFilter,
+        FilterInterface $productFilter,
+        AdjustmentInterface $promotionAdjustment1,
+        AdjustmentInterface $promotionAdjustment2,
+        Collection $originalItems,
+        Collection $units,
+        OrderInterface $order,
+        OrderItemInterface $orderItem1,
+        OrderItemUnitInterface $unit1,
+        OrderItemUnitInterface $unit2,
+        PromotionInterface $promotion,
+        CurrencyConverterInterface $currencyConverter
+    ) {
+        $order->getCurrencyCode()->willReturn('PLN');
+
+        $currencyConverter->convertToBase(1000, 'PLN')->willReturn(250);
+
+        $order->getItems()->willReturn($originalItems);
```

Comment:
`$order->getItems()->willReturn(new ArrayCollection([$orderItem1]));` will let you delete `Collection $originalItems` mock.